test/LoggerTest.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Horde Log package.
+ *
+ * This package is based on Zend_Log from the Zend Framework
+ * (http://framework.zend.com).  Both that package and this
+ * one were written by Mike Naberezny and Chuck Hagenbuch.
+ *
+ * @author     Rafael te Boekhorst <boekhorstb1@b1-systems.de>
+ * @category   Horde
+ * @license    http://www.horde.org/licenses/bsd BSD
+ * @package    Log
+ * @subpackage UnitTests
+ */
+
+namespace Horde\Log\Test;
+
+use Horde\Log\Logger;
+use PHPUnit\Framework\TestCase;
+
+use Psr\Log\LoggerInterface;
+
+use Horde\Util\HordeString;
+use Psr\Log\InvalidArgumentException;
+use Stringable;
+
+use Horde\Log\Filter\MessageFilter;
+use Horde\Log\Handler\MockHandler;
+use Horde\Log\Handler\SyslogHandler;
+use ReflectionClass;
+
+use Horde\Log\LogHandler;
+use Horde\Log\LogException;
+use Horde_Log;
+use Horde\Log\LogMessage;
+use Horde\Log\LogLevel;
+use Horde\Log\LogLevels;
+
+class LoggerTest extends TestCase
+{
+    public function setUp(): void
+    {
+        date_default_timezone_set('America/New_York');
+        $this->level1 = new LogLevel(Horde_Log::ALERT, 'Alert');
+        $this->message1 = 'this is an emergency!';
+        $this->message2 = 'this is a polar bear!';
+        $this->logMessage1 = new LogMessage($this->level1, $this->message1, ['timestamp' => date('c')]);
+        $this->logMessage2 = new LogMessage($this->level1, $this->message2);
+
+        $this->loglevelsss = new LogLevels();
+        $this->messagefilter[] = new MessageFilter('/emergency/');
+
+        $this->mockhandler1 = new MockHandler();
+        $this->mockhandler2 = new MockHandler();
+        $this->mockhandler2->addFilter(new MessageFilter('/notwokringregexpattern/'));
+        $this->handlers[] = $this->mockhandler1;
+        $this->handlers[] = $this->mockhandler2;
+
+
+        $this->logging = new Logger($this->handlers, null, $this->messagefilter);
+    }
+
+    public function testSerializeAndAddFilter(): void
+    {
+        $data = $this->logging->serialize();
+        $this->assertStringContainsString('MockHandler', $data);
+        $this->assertStringContainsString('MessageFilter', $data);
+        $this->assertStringContainsString('/emergency/', $data);
+    }
+
+    public function testSerializeAndAddHandler(): void
+    {
+        $mockhandler3 = new MockHandler();
+        $mockhandler3->addFilter(new MessageFilter('/whazzaaa/'));
+        $this->logging->addHandler($mockhandler3);
+        $data = $this->logging->serialize();
+        $this->assertStringContainsString('/whazzaaa/', $data);
+    }
+
+    public function testUnserialize(): void
+    {
+        $data = $this->logging->serialize();
+        $this->assertNull($this->logging->unserialize($data));
```

Comment:
Testing that unserialize returns null is not very helpful. Maybe create a new logger object and unserialize the data of the first one. Then check if the handlers/filters are the same.

Reply:
Ok I will change this accordingly